crates/sui-core/src/authority/authority_store.rs

```diff
@@ -248,15 +248,14 @@ impl AuthorityStore {
             .contains_key(digest)?)
     }
 
-    pub fn transaction_executed_in_epoch(
+    pub fn get_transaction_checkpoint(
         &self,
         digest: &TransactionDigest,
-    ) -> SuiResult<Option<EpochId>> {
+    ) -> SuiResult<Option<(EpochId, CheckpointSequenceNumber)>> {
         Ok(self
             .perpetual_tables
             .executed_transactions_to_checkpoint
-            .get(digest)?
-            .map(|(epoch, _)| epoch))
+            .get(digest)?)
```

Comment:
last week we discussed whether it would be possible to prune this table / do without it. If we can't do without it, we need to include this table in snapshots. (Also, it grows monotonically at something like 4GB per year per transaction-per-second.)

So:
- will it be okay for nodes to prune this data and return an error for old txns? (seems okay to me)
- Will it be okay for recently-loaded-from-snapshot nodes to return an error for very recent txns here?

Reply:
This is an optional field, so it's ok to be None. I think in general for a query to a fullnode we should expect it has limited history. But we should also expect some nodes have longer history so that some explorers could query as well